TST: coordinates reshape/ravel tests fail with remote data

Example log: https://travis-ci.org/github/astropy/astropy/jobs/680388850

```
_________________________ TestManipulation.test_ravel __________________________
self = <astropy.coordinates.tests.test_shape_manipulation.TestManipulation object at 0x7f1a5e7c1630>
    def test_ravel(self):
        s0_ravel = self.s0.ravel()
        assert s0_ravel.shape == (self.s0.size,)
        assert np.all(s0_ravel.data.lon == self.s0.data.lon.ravel())
        assert np.may_share_memory(s0_ravel.data.lon, self.s0.data.lon)
        assert np.may_share_memory(s0_ravel.data.lat, self.s0.data.lat)
        # Since s1 lon, lat were broadcast, ravel needs to make a copy.
        s1_ravel = self.s1.ravel()
        assert s1_ravel.shape == (self.s1.size,)
        assert np.all(s1_ravel.data.lon == self.s1.data.lon.ravel())
>       assert not np.may_share_memory(s1_ravel.data.lat, self.s1.data.lat)
E       AssertionError: assert not True
E        +  where True = <function may_share_memory at 0x7f1a7b482048>(<Latitude [-90., -60., -30.,   0.,  30.,  60.,  90., -90., -60., -30.,\n             0.,  30.,  60.,  90., -90., -60., ....,  90., -90., -60.,\n           -30.,   0.,  30.,  60.,  90., -90., -60., -30.,   0.,  30.,\n            60.,  90.] deg>, <Latitude [[-90., -60., -30.,   0.,  30.,  60.,  90.],\n           [-90., -60., -30.,   0.,  30.,  60.,  90.],\n        ...0.],\n           [-90., -60., -30.,   0.,  30.,  60.,  90.],\n           [-90., -60., -30.,   0.,  30.,  60.,  90.]] deg>)
E        +    where <function may_share_memory at 0x7f1a7b482048> = np.may_share_memory
E        +    and   <Latitude [-90., -60., -30.,   0.,  30.,  60.,  90., -90., -60., -30.,\n             0.,  30.,  60.,  90., -90., -60., ....,  90., -90., -60.,\n           -30.,   0.,  30.,  60.,  90., -90., -60., -30.,   0.,  30.,\n            60.,  90.] deg> = <UnitSphericalRepresentation (lon, lat) in (hourangle, deg)\n    [( 0., -90.), ( 0., -60.), ( 0., -30.), ( 0.,   0.), (...0.), (16.,  90.),\n     (20., -90.), (20., -60.), (20., -30.), (20.,   0.), (20.,  30.),\n     (20.,  60.), (20.,  90.)]>.lat
E        +      where <UnitSphericalRepresentation (lon, lat) in (hourangle, deg)\n    [( 0., -90.), ( 0., -60.), ( 0., -30.), ( 0.,   0.), (...0.), (16.,  90.),\n     (20., -90.), (20., -60.), (20., -30.), (20.,   0.), (20.,  30.),\n     (20.,  60.), (20.,  90.)]> = <AltAz Coordinate (obstime=['2012-01-01 00:00:00.000' '2012-01-01 00:00:00.000'\n '2012-01-01 00:00:00.000' '2012-01-01...0.,  90.),\n     (300., -90.), (300., -60.), (300., -30.), (300.,   0.), (300.,  30.),\n     (300.,  60.), (300.,  90.)]>.data
E        +    and   <Latitude [[-90., -60., -30.,   0.,  30.,  60.,  90.],\n           [-90., -60., -30.,   0.,  30.,  60.,  90.],\n        ...0.],\n           [-90., -60., -30.,   0.,  30.,  60.,  90.],\n           [-90., -60., -30.,   0.,  30.,  60.,  90.]] deg> = <UnitSphericalRepresentation (lon, lat) in (hourangle, deg)\n    [[( 0., -90.), ( 0., -60.), ( 0., -30.), ( 0.,   0.), ... (16.,  90.)],\n     [(20., -90.), (20., -60.), (20., -30.), (20.,   0.), (20.,  30.),\n      (20.,  60.), (20.,  90.)]]>.lat
E        +      where <UnitSphericalRepresentation (lon, lat) in (hourangle, deg)\n    [[( 0., -90.), ( 0., -60.), ( 0., -30.), ( 0.,   0.), ... (16.,  90.)],\n     [(20., -90.), (20., -60.), (20., -30.), (20.,   0.), (20.,  30.),\n      (20.,  60.), (20.,  90.)]]> = <AltAz Coordinate (obstime=[['2012-01-01 00:00:00.000' '2012-01-01 00:00:00.000'\n  '2012-01-01 00:00:00.000' '2012-01-... 90.)],\n     [(300., -90.), (300., -60.), (300., -30.), (300.,   0.),\n      (300.,  30.), (300.,  60.), (300.,  90.)]]>.data
E        +        where <AltAz Coordinate (obstime=[['2012-01-01 00:00:00.000' '2012-01-01 00:00:00.000'\n  '2012-01-01 00:00:00.000' '2012-01-... 90.)],\n     [(300., -90.), (300., -60.), (300., -30.), (300.,   0.),\n      (300.,  30.), (300.,  60.), (300.,  90.)]]> = <astropy.coordinates.tests.test_shape_manipulation.TestManipulation object at 0x7f1a5e7c1630>.s1
../../.tox/py37-test-devdeps/lib/python3.7/site-packages/astropy/coordinates/tests/test_shape_manipulation.py:123: AssertionError
________________________ TestManipulation.test_reshape _________________________
self = <astropy.coordinates.tests.test_shape_manipulation.TestManipulation object at 0x7f1a5f462be0>
    def test_reshape(self):
        s0_reshape = self.s0.reshape(2, 3, 7)
        assert s0_reshape.shape == (2, 3, 7)
        assert np.all(s0_reshape.data.lon == self.s0.data.lon.reshape(2, 3, 7))
        assert np.all(s0_reshape.data.lat == self.s0.data.lat.reshape(2, 3, 7))
        assert np.may_share_memory(s0_reshape.data.lon, self.s0.data.lon)
        assert np.may_share_memory(s0_reshape.data.lat, self.s0.data.lat)
        s1_reshape = self.s1.reshape(3, 2, 7)
        assert s1_reshape.shape == (3, 2, 7)
        assert np.all(s1_reshape.data.lat == self.s1.data.lat.reshape(3, 2, 7))
        assert np.may_share_memory(s1_reshape.data.lat, self.s1.data.lat)
        assert np.all(s1_reshape.obstime == self.s1.obstime.reshape(3, 2, 7))
        assert np.may_share_memory(s1_reshape.obstime.jd1,
                                   self.s1.obstime.jd1)
        assert np.all(s1_reshape.location == self.s1.location.reshape(3, 2, 7))
        assert np.may_share_memory(s1_reshape.location, self.s1.location)
        assert np.all(s1_reshape.temperature ==
                      self.s1.temperature.reshape(3, 2, 7))
        assert np.may_share_memory(s1_reshape.temperature,
                                   self.s1.temperature)
        assert s1_reshape.pressure == self.s1.pressure
        # For reshape(3, 14), copying is necessary for lon, lat, location, time
        s1_reshape2 = self.s1.reshape(3, 14)
        assert s1_reshape2.shape == (3, 14)
        assert np.all(s1_reshape2.data.lon == self.s1.data.lon.reshape(3, 14))
>       assert not np.may_share_memory(s1_reshape2.data.lon, self.s1.data.lon)
E       AssertionError: assert not True
E        +  where True = <function may_share_memory at 0x7f1a7b482048>(<Longitude [[ 0.,  0.,  0.,  0.,  0.,  0.,  0.,  4.,  4.,  4.,  4.,  4.,\n              4.,  4.],\n            [ 8.,  8.... 12., 12.],\n            [16., 16., 16., 16., 16., 16., 16., 20., 20., 20., 20., 20.,\n             20., 20.]] hourangle>, <Longitude [[ 0.,  0.,  0.,  0.,  0.,  0.,  0.],\n            [ 4.,  4.,  4.,  4.,  4.,  4.,  4.],\n            [ 8.,  8...12., 12.],\n            [16., 16., 16., 16., 16., 16., 16.],\n            [20., 20., 20., 20., 20., 20., 20.]] hourangle>)
E        +    where <function may_share_memory at 0x7f1a7b482048> = np.may_share_memory
E        +    and   <Longitude [[ 0.,  0.,  0.,  0.,  0.,  0.,  0.,  4.,  4.,  4.,  4.,  4.,\n              4.,  4.],\n            [ 8.,  8.... 12., 12.],\n            [16., 16., 16., 16., 16., 16., 16., 20., 20., 20., 20., 20.,\n             20., 20.]] hourangle> = <UnitSphericalRepresentation (lon, lat) in (hourangle, deg)\n    [[( 0., -90.), ( 0., -60.), ( 0., -30.), ( 0.,   0.), ...  60.), (16.,  90.), (20., -90.), (20., -60.), (20., -30.),\n      (20.,   0.), (20.,  30.), (20.,  60.), (20.,  90.)]]>.lon
E        +      where <UnitSphericalRepresentation (lon, lat) in (hourangle, deg)\n    [[( 0., -90.), ( 0., -60.), ( 0., -30.), ( 0.,   0.), ...  60.), (16.,  90.), (20., -90.), (20., -60.), (20., -30.),\n      (20.,   0.), (20.,  30.), (20.,  60.), (20.,  90.)]]> = <AltAz Coordinate (obstime=[['2012-01-01 00:00:00.000' '2012-01-01 00:00:00.000'\n  '2012-01-01 00:00:00.000' '2012-01-...  90.), (300., -90.),\n      (300., -60.), (300., -30.), (300.,   0.), (300.,  30.),\n      (300.,  60.), (300.,  90.)]]>.data
E        +    and   <Longitude [[ 0.,  0.,  0.,  0.,  0.,  0.,  0.],\n            [ 4.,  4.,  4.,  4.,  4.,  4.,  4.],\n            [ 8.,  8...12., 12.],\n            [16., 16., 16., 16., 16., 16., 16.],\n            [20., 20., 20., 20., 20., 20., 20.]] hourangle> = <UnitSphericalRepresentation (lon, lat) in (hourangle, deg)\n    [[( 0., -90.), ( 0., -60.), ( 0., -30.), ( 0.,   0.), ... (16.,  90.)],\n     [(20., -90.), (20., -60.), (20., -30.), (20.,   0.), (20.,  30.),\n      (20.,  60.), (20.,  90.)]]>.lon
E        +      where <UnitSphericalRepresentation (lon, lat) in (hourangle, deg)\n    [[( 0., -90.), ( 0., -60.), ( 0., -30.), ( 0.,   0.), ... (16.,  90.)],\n     [(20., -90.), (20., -60.), (20., -30.), (20.,   0.), (20.,  30.),\n      (20.,  60.), (20.,  90.)]]> = <AltAz Coordinate (obstime=[['2012-01-01 00:00:00.000' '2012-01-01 00:00:00.000'\n  '2012-01-01 00:00:00.000' '2012-01-... 90.)],\n     [(300., -90.), (300., -60.), (300., -30.), (300.,   0.),\n      (300.,  30.), (300.,  60.), (300.,  90.)]]>.data
E        +        where <AltAz Coordinate (obstime=[['2012-01-01 00:00:00.000' '2012-01-01 00:00:00.000'\n  '2012-01-01 00:00:00.000' '2012-01-... 90.)],\n     [(300., -90.), (300., -60.), (300., -30.), (300.,   0.),\n      (300.,  30.), (300.,  60.), (300.,  90.)]]> = <astropy.coordinates.tests.test_shape_manipulation.TestManipulation object at 0x7f1a5f462be0>.s1
../../.tox/py37-test-devdeps/lib/python3.7/site-packages/astropy/coordinates/tests/test_shape_manipulation.py:257: AssertionError
```

Basically things that are tested for not sharing memory now share memory. Seems concerning?

cc @mhvk

--- a/astropy/coordinates/tests/test_shape_manipulation.py
+++ b/astropy/coordinates/tests/test_shape_manipulation.py
@@ -22,11 +22,13 @@ class TestManipulation():
     """
 
     def setup(self):
+        # For these tests, we set up frames and coordinates using copy=False,
+        # so we can check that broadcasting is handled correctly.
         lon = Longitude(np.arange(0, 24, 4), u.hourangle)
         lat = Latitude(np.arange(-90, 91, 30), u.deg)
-        # With same-sized arrays, no attributes
+        # With same-sized arrays, no attributes.
         self.s0 = ICRS(lon[:, np.newaxis] * np.ones(lat.shape),
-                       lat * np.ones(lon.shape)[:, np.newaxis])
+                       lat * np.ones(lon.shape)[:, np.newaxis], copy=False)
         # Make an AltAz frame since that has many types of attributes.
         # Match one axis with times.
         self.obstime = (Time('2012-01-01') +
@@ -42,7 +44,7 @@ class TestManipulation():
                         obstime=self.obstime,
                         location=self.location,
                         pressure=self.pressure,
-                        temperature=self.temperature)
+                        temperature=self.temperature, copy=False)
         # For some tests, also try a GCRS, since that has representation
         # attributes.  We match the second dimension (via the location)
         self.obsgeoloc, self.obsgeovel = self.location.get_gcrs_posvel(
@@ -50,13 +52,14 @@ class TestManipulation():
         self.s2 = GCRS(ra=lon[:, np.newaxis], dec=lat,
                        obstime=self.obstime,
                        obsgeoloc=self.obsgeoloc,
-                       obsgeovel=self.obsgeovel)
+                       obsgeovel=self.obsgeovel, copy=False)
         # For completeness, also some tests on an empty frame.
         self.s3 = GCRS(obstime=self.obstime,
                        obsgeoloc=self.obsgeoloc,
-                       obsgeovel=self.obsgeovel)
+                       obsgeovel=self.obsgeovel, copy=False)
         # And make a SkyCoord
-        self.sc = SkyCoord(ra=lon[:, np.newaxis], dec=lat, frame=self.s3)
+        self.sc = SkyCoord(ra=lon[:, np.newaxis], dec=lat, frame=self.s3,
+                           copy=False)
 
     def test_getitem0101(self):
         # We on purpose take a slice with only one element, as for the

